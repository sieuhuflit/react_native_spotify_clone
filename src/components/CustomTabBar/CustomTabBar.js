import React from 'react';
import PropTypes from 'prop-types';
import { BottomTabBar } from 'react-navigation';
import BarMusicPlayer from '../BarMusicPlayer';

const CustomTabBar = props => {
  const {
    screenProps: { currentSongData }
  } = props;

  return (
    <React.Fragment>
      {currentSongData && <BarMusicPlayer song={currentSongData} />}
      <BottomTabBar {...props} />
    </React.Fragment>
  );
};

CustomTabBar.propTypes = {
  navigation: PropTypes.object.isRequired,
  screenProps: PropTypes.object.isRequired
};

export default CustomTabBar;
