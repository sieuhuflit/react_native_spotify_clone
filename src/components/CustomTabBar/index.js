import CustomTabBar from './CustomTabBar';

export default CustomTabBar;
