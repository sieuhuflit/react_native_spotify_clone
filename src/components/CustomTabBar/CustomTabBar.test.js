import React from 'react';
import { shallow } from 'enzyme';
import CustomTabBar from './CustomTabBar';

const setup = propOverrides => {
  const props = Object.assign(
    {
      screenProps: {
        customSongData: {
          name: 'hi'
        }
      }
    },
    propOverrides
  );
  const wrapper = shallow(<CustomTabBar {...props} />);
  return {
    props,
    wrapper
  };
};

describe('CustomTabBar check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
