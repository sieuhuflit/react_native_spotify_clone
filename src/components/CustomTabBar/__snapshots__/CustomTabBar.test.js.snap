// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabBar check snapshot should match snapshot 1`] = `ShallowWrapper {}`;
